test/e2e/ingress.go

```diff
@@ -1027,3 +1027,129 @@ var _______ = describe("Ingress tests simple NLB", func() {
+		jig *ingress.TestJig
+	)
+
+	It("Should activate OPA filter without issue [Ingress] [Opa]", func() {
+		jig = ingress.NewIngressTestJig(f.ClientSet)
+		cs = f.ClientSet
+		serviceName := "styra-smoketest"
+		ns := f.Namespace.Name
+		hostName := fmt.Sprintf("%s-%d.%s", serviceName, time.Now().UTC().Unix(), E2EHostedZone())
+		labels := map[string]string{
+			"app": serviceName,
+		}
+		port := 8080
+		replicas := int32(3)
+		targetPort := 9090
+		backendContent := "mytest"
+		route := fmt.Sprintf(`* -> inlineContent("%s") -> <shunt>`, backendContent)
+		waitTime := 10 * time.Minute
+
+		// CREATE setup
+		// backend deployment
+		By("Creating a deployment with " + serviceName + " in namespace " + ns)
+		depl := createSkipperBackendDeployment(serviceName, ns, route, labels, int32(targetPort), replicas)
+		_, err := cs.AppsV1().Deployments(ns).Create(context.TODO(), depl, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		By("Creating service " + serviceName + " in namespace " + ns)
+		service := createServiceTypeClusterIP(serviceName, labels, port, targetPort)
+		_, err = cs.CoreV1().Services(ns).Create(context.TODO(), service, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ing := createIngress(serviceName, hostName, ns, "/", netv1.PathTypeImplementationSpecific, labels, nil, port)
+		ingressCreate, err := cs.NetworkingV1().Ingresses(ns).Create(context.TODO(), ing, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		addr, err := jig.WaitForIngressAddress(context.TODO(), cs, ns, ingressCreate.Name, waitTime)
+		framework.ExpectNoError(err)
+
+		_, err = cs.NetworkingV1().Ingresses(ns).Get(context.TODO(), ing.Name, metav1.GetOptions{ResourceVersion: "0"})
+		framework.ExpectNoError(err)
+
+		// skipper http -> https redirect
+		By("Waiting for skipper route to default redirect from http to https, to see that our ingress-controller and skipper works")
+		err = waitForResponse(addr, "http", waitTime, isRedirect, true)
+		framework.ExpectNoError(err)
+
+		// LB ready
+		By("Waiting for ALB to create endpoint " + addr + " and skipper route, to see that our ingress-controller and skipper works")
+		err = waitForResponse(addr, "https", waitTime, isNotFound, true)
+		framework.ExpectNoError(err)
+
+		// DNS ready
+		By("Waiting for DNS to see that external-dns and skipper route to service and pod works")
+		err = waitForResponse(hostName, "https", waitTime, isSuccess, false)
+		framework.ExpectNoError(err)
+
+		// Test that we get content from the default ingress
+		By("By checking the content of the reply we see that the ingress stack works")
+		rt, quit := createHTTPRoundTripper()
+		defer func() {
+			quit <- struct{}{}
+		}()
+		url := "https://" + hostName + "/"
+		req, err := http.NewRequest("GET", url, nil)
+		framework.ExpectNoError(err)
+		resp, err := rt.RoundTrip(req)
+		framework.ExpectNoError(err)
+		s, err := getBody(resp)
+		framework.ExpectNoError(err)
+		Expect(s).To(Equal(backendContent))
```

Comment:
This appears to be identical to the testcase above. It would be best to "combine" both test cases so that we only do the setup once and use it for both tests. 

This is, I believe, also adding a lot to the test running time because this process would take a few minutes to complete.

Reply:
You can check https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/test/e2e/authorization.go for inspiration on how to design the test suite and extract common setup.

Reply:
The test above `It("Should create simple NLB ingress [Ingress]"`, (or any other Ingress related test for that matter) are not executed in the pipeline e2e testing AFAIU. I came to this conclusion looking at the FOCUS here https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/test/e2e/run_e2e.sh#L186
Also the junit report confirms this. If that is the case, it does not add time to the pipeline time anyway.
```
<testcase name="[It] [zalando] Ingress tests simple NLB Should create simple NLB ingress [Ingress]" classname="Kubernetes e2e suite" status="skipped" time="0">
              <skipped message="skipped"></skipped>
```

We would actually like to have the OPA related tests run during the e2e step in the pipeline. So I feel like it's good to keep it separated from others.